nccl/common/nccl-common.sh

```diff
@@ -221,31 +221,34 @@ prepare_instance() {
         num_instances=$2
         INSTANCES=()
         create_pg
-        while [ ${num_instances} -gt 0 ] ; do
-            create_instance_attempts=0
-            INSTANCE_STATE="unavailable"
-            while [ ${INSTANCE_STATE} != 'running' ] && [ ${create_instance_attempts} -lt ${create_instance_retries} ] ; do
-                if [ $1 == 'ami_instance' ] ; then
-                    create_instance ${SSHSG} 1 ${prep_ami} ${instance_ami_type}
-                else
-                    create_instance ${SGId} 1 ${AMIS["${AWS_DEFAULT_REGION}"]} ${instance_test_type}
-                fi
-                if [ ${create_instance_exit_code} -ne 0 ]; then
-                    echo "==> Changing the region"
-                    # Start over with new region
```

Comment:
In the commit message, we say `...avoid launching instances in different subnets`. I think changing to `region` in commit message is more appropriate.

Reply:
We will still launch in a different region if capacity is low in the current region. Previously we were launching one instance at a time, and in the same region switching to different AZ if there was low capacity in a particular AZ.

Reply:
And now, we ensure that all instances are either launched in same AZ, same region, or failed. Right?

Reply:
Yes, even if one instance fails to launch we terminate all instances and try again in a different AZ